pandas/_libs/tslibs/conversion.pyx

```diff
@@ -888,39 +887,43 @@ def tz_localize_to_utc(ndarray[int64_t] vals, object tz, object ambiguous=None,
                              "the same size as vals")
         ambiguous_array = np.asarray(ambiguous)
 
+    if is_string_object(nonexistent):
+        if nonexistent == 'NaT':
+            fill_nonexist = True
+        elif nonexistent == 'shift':
+            shift = True
+
```

Comment:
in theory should assert the possible values of nonexistent (e.g. what If I pass 'foo'),